doxygen/doxygen.bzl

```diff
@@ -18,37 +18,36 @@ def _swift_doxygen_impl(ctx):
     doxygen_out = ctx.actions.declare_directory(ctx.attr.name + "_doxygen")
     vars["DOXYGEN_OUTPUT_DIRECTORY"] = doxygen_out.path
 
+    # this performs a CMake-like replacement of @VAR@ based on the vars dict
     config = configure_file_impl(ctx, vars, ctx.attr.name + "_Doxyfile")[0].files.to_list()[0]
 
-    input_dir = ""
-    if "PROJECT_SOURCE_DIR" in vars:
-        input_dir = vars["PROJECT_SOURCE_DIR"]
-
-    project_name = ""
-    if "PROJECT_NAME" in vars:
-        project_name = vars["PROJECT_NAME"]
-
     ctx.actions.run_shell(
         inputs = [config] + ctx.files.deps,
         outputs = [doxygen_out],
+        env = vars,
         command = """
         DOXYGEN_DOT_FOUND=NO
         DOXYGEN_DOT_PATH=
-
         if command -v dot &> /dev/null
         then
             DOXYGEN_DOT_FOUND=YES
             DOXYGEN_DOT_PATH=$(which dot)
         fi
 
+        PRW=`pwd`
+
+        # backward compatibility with old CMake-style doxygen config files
```

Comment:
Do we still use old doxygen config files?

Reply:
What I meant here was the use of `@VAR@` in the config files. Unfortunately, I had to maintain those in the new Doxygen config files. Otherwise the CMake builds would fail. So I was either maintain it here (for the moment) or have duplicate config files, for CMake and Bazel. That didn't seem like the best approach. I hope we can drop CMake at some point and use the default Doxygen approach `$(VAR)`.